lib/rucio/web/ui/flask/common/utils.py

```diff
@@ -0,0 +1,626 @@
+                ui_account, ui_vo = select_account_name(saml_nameid, 'saml', ui_vo)
+            elif ui_account is None:
+                ui_account, _ = select_account_name(saml_nameid, 'saml', ui_vo)
+            elif ui_vo is None:
+                _, ui_vo = select_account_name(saml_nameid, 'saml', ui_vo)
+
+            # Try to eliminate VOs based on the account name (if we have one), if we still have multiple options let the user select one
+            if type(ui_vo) is list:
+                if ui_account:
+                    valid_vos = []
+                    for vo in ui_vo:
+                        if account_exists(ui_account, vo):
+                            valid_vos.append(vo)
+                    if len(valid_vos) == 0:
+                        return render_template("problem.html", msg=('Cannot find any Rucio account %s associated with identity %s at any VO.' % (html_escape(ui_account), html_escape(saml_nameid))))
+                    elif len(valid_vos) == 1:
+                        ui_vo = valid_vos[0]
+                    else:
+                        vos_with_desc = get_vo_descriptions(valid_vos)
+                        return add_cookies(make_response(render_template("select_login_method.html", oidc_issuers=AUTH_ISSUERS, saml_support=SAML_SUPPORT, possible_vos=vos_with_desc)))
+                else:
+                    vos_with_desc = get_vo_descriptions(ui_vo)
+                    return add_cookies(make_response(render_template("select_login_method.html", oidc_issuers=AUTH_ISSUERS, saml_support=SAML_SUPPORT, possible_vos=vos_with_desc)))
+
+            if not ui_account:
+                if MULTI_VO:
+                    return render_template("problem.html", msg='Cannot get find any account associated with %s identity at VO %s.' % (html_escape(saml_nameid), html_escape(ui_vo)))
+                else:
+                    return render_template("problem.html", msg='Cannot get find any account associated with %s identity.' % (html_escape(saml_nameid)))
+            token = get_token(auth.get_auth_token_saml, acc=ui_account, vo=ui_vo, idt=saml_nameid)
+            if not token:
+                if MULTI_VO:
+                    return render_template("problem.html", msg=('Cannot get auth token. It is possible that the presented identity %s is not mapped to any Rucio account %s at VO %s.') % (html_escape(saml_nameid), html_escape(ui_account), html_escape(ui_vo)))
+                else:
+                    return render_template("problem.html", msg=('Cannot get auth token. It is possible that the presented identity %s is not mapped to any Rucio account %s.') % (html_escape(saml_nameid), html_escape(ui_account)))
+            return finalize_auth(token, 'saml', cookie_extra)
+
+        return render_template("problem.html", msg="Not authenticated")
+
+    return render_template("problem.html", msg="Error while processing SAML")
+
+
+def oidc_auth(account, issuer, ui_vo=None):
+    """
+    # TODO: Validate for Flask
```

Comment:
I guess this should be done before the merge

Reply:
Same as above.